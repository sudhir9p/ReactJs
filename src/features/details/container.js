import React from 'react';
import './details.css';

export class MoviesDetails extends React.Component {
    constructor(props) {

        super(props);
        this.movie = this.props.location.data.movie;
    }

    render() {
        return (
            <div className="movie-details">
                <div className="row">
                    <div className="col-sm-3 details-image">
                        <img src={this.movie.poster_path} />
                    </div>
                    <div className="col-sm-6 details-content">
                        <div class="row">
                            <h2 className="details-title" >{this.movie.title}</h2> <span className="details-rating">4.3</span>
                        </div>
                        <div class="row details-yearmin">
                            <p class="details-year">{this.movie.year}</p> <p class="details-year-text">year</p> <p class="details-mins">{this.movie.runtime}</p> <p class="details-mins-text">mins</p>
                        </div>
                        <div className="row details-movie-desc">
                            <p>{this.movie.overview}</p>
                        </div>

                    </div>
                </div>
            </div>
        )
    }
}

export default MoviesDetails;