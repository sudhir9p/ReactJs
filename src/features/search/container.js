import React from 'react';
import SearchForm from './search-form/index';
import { connect } from "react-redux";
import { fetchMovies, searchMovies, movieSearchText } from '../../core/state/actions';
import { configuration } from '../../../config/config.json'
import SearchResult from './search-results';
import { stringFormat } from './utils';

export class Search extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            data: []
        }
    }

    componentDidMount = () => {
        this.getMovies(configuration.apiUrl).then(res => {
            this.setState({ data: res.data });

        });
    }


    async componentDidUpdate(prevProps, prevState, snapshot) {
        if (prevProps.searchBy != this.props.searchBy || prevProps.sortBy != this.props.searchBy) {
            const search = this.props.searchBy ? this.props.searchBy : "";
            const sortBy = this.props.sortBy ? this.props.sortBy : "";
            const movies = await this.getMovies(stringFormat(configuration.apiSortUrl, sortBy, "desc", search));
            this.setState({ data: movies.data });
        }
    }


    getMovies = (url) => {
        return fetch(url).then(data => {
            return data.json()
        });
    }

    render = () => {
        return (
            <div className="searchForm">
                <SearchForm></SearchForm>
                <SearchResult data={this.state.data} history={this.props.history}  onMovieClick={this.props.onMovieClick}></SearchResult>
            </div>
        );
    }

}



//export default Search;
const mapStateToProps = state => ({
    sortBy: state.sortBy,
    searchBy: state.searchBy
});

export default connect(mapStateToProps)(Search)