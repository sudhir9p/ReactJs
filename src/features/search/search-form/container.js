import React from 'react';
import { connect } from "react-redux";
import { searchMovies, sortMovies } from '../../../core/state/actions';
import { SearchFormComponent } from './component'

export class SearchForm extends React.Component {
    constructor(props) {
        super(props);
    }

    onSearchClick = () => {
        if (this.state.searchText) {
            this.props.searchByMap(this.state.searchText);
        }
    }

    onSortBy = (sortBy) => {
        this.props.sortByMap(sortBy);
    }

    onSearchTextChange = (e) => {
        this.setState({ searchText: e.target.value });
    }


    render() {
        return (
            <SearchFormComponent data={this.props.data} onSearchTextChange={this.onSearchTextChange} history={this.props.history} onSortBy={this.onSortBy} onMovieClick={this.props.onMovieClick}></SearchFormComponent>
        )
    }

}

const mapStateToProps = state => ({
    searchBy: state.searchBy,
    sortBy: state.SortBy
});


const mapDispatchToProps = dispatch => ({
    searchByMap: (value) => {
        dispatch(searchMovies(value));
    },
    sortByMap: (value) => {
        dispatch(sortMovies(value));
    }
});



export default connect(mapStateToProps, mapDispatchToProps)(SearchForm)