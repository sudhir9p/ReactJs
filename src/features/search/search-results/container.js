import React from 'react';
import MoviesList from '../../../shared/movie-list';

export class SearchResult extends React.Component {
    constructor(props) {
        super(props);
    }


    onMovieClick = (movie) => {
        this.props.history.push({
            pathname: `/details/${movie.id}`
        });
    }

    render = () => {
        console.log('BBBBBBBBBB');
        console.log(this.props);
        return (
            <MoviesList movies={this.props.data ? this.props.data : []} history={this.props.history} onMovieClick={this.onMovieClick}></MoviesList>
        );

    }

}

export default SearchResult;

