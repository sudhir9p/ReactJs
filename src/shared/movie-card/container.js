import React from 'react';

export class MoviesCard extends React.Component {
    constructor(props) {
        super(props);
    }
    render() {
        const movie = this.props.movie;
        return (
            <div className="movie-card" onClick={() => this.props.onMovieClick(movie)}>
                <img src={movie.poster_path} alt={movie.title} />
                <p className="txtcolor">{movie.title}</p>
                <p className="bold">Release Date: <span className="normal">{movie.release_date}</span></p>
                <p className="bold">Rating: <span className="normal">{movie.vote_count}</span></p>
                <p className="bold">Genre: <span className="normal">{movie.genres.join(", ")}</span></p>
            </div>
        )
    }
}

export default MoviesCard;