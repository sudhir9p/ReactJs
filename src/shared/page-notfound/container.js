import React from 'react';
import { withRouter } from 'react-router';

export const PageNotFound = (props) => {
    return (
        <div className="movie-list-container">
            <h1>Page Not Found</h1>
        </div>
    )
}

export default withRouter(PageNotFound);