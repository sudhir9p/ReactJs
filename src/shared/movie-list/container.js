import React from 'react';
import { MoviesCard } from '../movie-card/index';
import { withRouter } from 'react-router';

export class MoviesList extends React.Component {

    onMovieClick = (movie) => {
        this.props.history.push({
            pathname: '/details',
            data: {
                movie
            }
        });
    }

    render() {
        return (
            <div className="movie-list-container">
                {this.props.movies.map(movie => <div key={movie.id} onClick={() => this.onMovieClick(movie)}> <MoviesCard movie={movie}  /></div>)}
            </div>
        )
    }
}

export default withRouter(MoviesList);