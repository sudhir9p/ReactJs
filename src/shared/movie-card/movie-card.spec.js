import { MoviesCard } from './container';
import React from 'react';
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe(`Movie-Card Component`, () => {
    const movie = {
        "id": 337167,
        "title": "Fifty Shades Freed",
        "tagline": "Don't miss the climax",
        "vote_average": 6.1,
        "vote_count": 1195,
        "release_date": "2018-02-07",
        "poster_path": "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg",
        "overview": "Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.",
        "budget": 55000000,
        "revenue": 136906000,
        "genres": [
            "Drama",
            "Romance"
        ],
        "runtime": 106,
        "year": 1992
    };

    it('should render as expected', () => {
        const movieCardOj = renderer.create(<MoviesCard movie={movie} />).toJSON();
        expect(movieCardOj).toMatchSnapshot();
    });


    it("Should trigger search on click", () => {
        const component = shallow(<MoviesCard movie={movie}/>);
        expect(component).toBeTruthy();

    })

})