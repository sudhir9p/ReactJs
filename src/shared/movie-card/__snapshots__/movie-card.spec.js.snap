// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie-Card Component should render as expected 1`] = `
<div
  className="movie-card"
  onClick={[Function]}
>
  <img
    alt="Fifty Shades Freed"
    src="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
  />
  <p
    className="txtcolor"
  >
    Fifty Shades Freed
  </p>
  <p
    className="bold"
  >
    Release Date: 
    <span
      className="normal"
    >
      2018-02-07
    </span>
  </p>
  <p
    className="bold"
  >
    Rating: 
    <span
      className="normal"
    >
      1195
    </span>
  </p>
  <p
    className="bold"
  >
    Genre: 
    <span
      className="normal"
    >
      Drama, Romance
    </span>
  </p>
</div>
`;
